packages/react-dom/src/__tests__/renderSubtreeIntoContainer-test.js

```diff
@@ -333,21 +352,33 @@ describe('renderSubtreeIntoContainer', () => {
       }
     }
 
-    ReactDOM.render(<Parent value="foo" />, container);
+    const root = ReactDOMClient.createRoot(container);
+    await act(async () => {
+      root.render(<Parent value="foo" />);
+    });
     expect(portal2.textContent).toBe('foo');
   });
 
-  it('fails gracefully when mixing React 15 and 16', () => {
+  it('fails gracefully when mixing React 15 and 16', async () => {
     class C extends React.Component {
       render() {
         return <div />;
       }
     }
-    const c = ReactDOM.render(<C />, document.createElement('div'));
+
+    const container = document.createElement('div');
+    const root = ReactDOMClient.createRoot(container);
+    const componentRef = React.createRef();
+    await act(async () => {
+      root.render(<C ref={componentRef} />);
+    });
     // React 15 calls this:
     // https://github.com/facebook/react/blob/77b71fc3c4/src/renderers/dom/client/ReactMount.js#L478-L479
-    expect(() => {
-      c._reactInternalInstance._processChildContext({});
+    const component = componentRef.current;
```

Comment:
This doesn't work because `component._reactInternalInstance` is an empty object and the test is to call `_processChildContext({})` which doesn't exist on `component._reactInternalInstance`

Reply:
@rickhanlonii do you know of a way around this? or if there is a better approach? Or maybe we can delete this test? (its for compatibility between React 15 & 16)

Reply:
I don't know if we can delete it, but we shouldn't change it to `createRoot` since that didn't exist until 18.